File: 2026-06/2026-06-08/jet_lagged.py
def get_jet_lag_hours(departure_city, arrival_city, flight_duration, direction):
    CITIES_UTC_OFFSET = {
        "Los Angeles": -8,
        "New York": -5,
        "London": 0,
        "Istanbul": +3,
        "Dubai": +4,
        "Hong Kong": +8,
        "Tokyo": +9
    }
    timezone_difference = abs(CITIES_UTC_OFFSET[arrival_city] - CITIES_UTC_OFFSET[departure_city])
    direction_multiplier = 1.5 if direction == "east" else 1.0

    return timezone_difference + (flight_duration * 0.1) * direction_multiplier

# --- TEST SUITE ---

tests_text = r'''
1. get_jet_lag_hours("Istanbul", "Hong Kong", 10, "east") should return 6.5.
2. get_jet_lag_hours("London", "New York", 8, "west") should return 5.8.
3. get_jet_lag_hours("Hong Kong", "Tokyo", 4, "east") should return 1.6.
4. get_jet_lag_hours("Dubai", "London", 7, "west") should return 4.7.
5. get_jet_lag_hours("Los Angeles", "Hong Kong", 15, "west") should return 17.5.
6. get_jet_lag_hours("Tokyo", "Dubai", 9, "west") should return 5.9.
7. get_jet_lag_hours("New York", "Istanbul", 10, "east") should return 9.5.
'''

import re

tests_regex = re.compile(r"(?P<number>\d+)\.\s(?P<function_call>.+) should return (?P<output>.+?)\.?$", re.MULTILINE)
test_data = [item.groupdict() for item in re.finditer(tests_regex, tests_text)]

def run_tests(test_data):
    print("——————————————————————————",
        "\n🧪 Starting Verification...",
        "\n——————————————————————————")

    fail_count = 0

    for test in test_data:
        function_call_output = eval(test['function_call'])
        test_output = eval(test['output'])
        
        if function_call_output == test_output:
            print(f"{test['number']}.✅ PASS - Function Call:\n{test['function_call']}")
        else:
            print(f"{test['number']}.❌ FAIL - Function Call:\n{test['function_call']}\nExpected:\n{test_output}\nGot:\n{function_call_output}")
            fail_count += 1

        print("————————————————————————————")

    print(
        f"⚠️ WARNING: {fail_count}/{len(test_data)} tests FAILED." if fail_count
        else "🎉 SUCCESS: All tests PASSED."
    )

run_tests(test_data)

File: 06 June/06-08/jetLagged.js
function getJetLagHours(departureCity, arrivalCity, flightDuration, direction) {
  const CITIES_UTC_OFFSET = {
    "Los Angeles": -8,
    "New York": -5,
    "London": 0,
    "Istanbul": +3,
    "Dubai": +4,
    "Hong Kong": +8,
    "Tokyo": +9
  };
  const timezoneDifference = Math.abs(CITIES_UTC_OFFSET[arrivalCity] - CITIES_UTC_OFFSET[departureCity]);
  const directionMultiplier = direction === "east" ? 1.5 : 1.0;

  return timezoneDifference + (flightDuration * 0.1) * directionMultiplier;
}

// --- TEST SUITE ---

const testsText = String.raw`
1. getJetLagHours("Istanbul", "Hong Kong", 10, "east") should return 6.5.
2. getJetLagHours("London", "New York", 8, "west") should return 5.8.
3. getJetLagHours("Hong Kong", "Tokyo", 4, "east") should return 1.6.
4. getJetLagHours("Dubai", "London", 7, "west") should return 4.7.
5. getJetLagHours("Los Angeles", "Hong Kong", 15, "west") should return 17.5.
6. getJetLagHours("Tokyo", "Dubai", 9, "west") should return 5.9.
7. getJetLagHours("New York", "Istanbul", 10, "east") should return 9.5.
`;

const testsRegex = /(?<number>\d+)\.\s(?<functionCall>.+) should return (?<output>.+?)\.?$/gm;
const testData = [...testsText.matchAll(testsRegex).map(match => match.groups)];

function runTests(testData) {
  console.log("——————————————————————————",
            "\n🧪Starting Verification...",
            "\n——————————————————————————");

  let failCount = 0;
  
  testData.forEach(test => {
    const functionCallOutput = eval(test.functionCall);
    const testOutput = eval(`(${test.output})`);

    if (JSON.stringify(functionCallOutput) === JSON.stringify(testOutput)) {
      console.log(`${test.number}.✅PASS - Function Call: ${test.functionCall}`);
    } else {
      console.log(`${test.number}.❌FAIL - Function Call: ${test.functionCall}\nExpected: ${JSON.stringify(testOutput)}\nGot: ${JSON.stringify(functionCallOutput)}`);

      ++failCount;
    }
    console.log("————————————————————————————");
  })

  console.log(failCount
    ? `⚠️WARNING: ${failCount}/${testData.length} tests FAILED.`
    : "🎉SUCCESS: All tests PASSED."
  );
}

runTests(testData);